cdap-security/src/main/java/co/cask/cdap/security/guice/SecureStoreModules.java

```diff
@@ -105,10 +117,18 @@ private SecureStoreManagerProvider(final CConfiguration cConf, Injector injector
      */
     @Override
     public SecureStoreManager get() {
-      if ("file".equalsIgnoreCase(cConf.get(Constants.Security.Store.PROVIDER))) {
-        return injector.getInstance(FileSecureStore.class);
+      if (AbstractSecureStore.isKMSBacked(cConf)) {
+        try {
+          // Check if required KMS classes are present.
+          Class.forName("org.apache.hadoop.crypto.key.kms.KMSClientProvider");
+          LOG.warn("Could not find classes required for supporting KMS provider.");
```

Comment:
ditto. actually, this code is the same as above, right? is it possible to share?